sources/engine/Stride.Assets/Entities/ComponentChecks/RequiredMembersCheck.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Reflection;
+using Stride.Core.Annotations;
+using Stride.Core.Assets;
+using Stride.Core.Assets.Compiler;
+using Stride.Engine;
+
+namespace Stride.Assets.Entities.ComponentChecks
+{
+    /// <summary>
+    /// Checks if all members of a component with a <see cref="MemberRequiredAttribute"/> are assigned a value.
+    /// </summary>
+    public class RequiredMembersCheck : IEntityComponentCheck
+    {
+        /// <inheritdoc/>
+        public bool AppliesTo(Type componentType)
+        {
+            return true; // applies to any component
+        }
+
+        /// <inheritdoc/>
+        public void Check(EntityComponent component, Entity entity, AssetItem assetItem, string targetUrlInStorage, AssetCompilerResult result)
+        {
+            var type = component.GetType();
+            var componentName = type.Name; // QUESTION: Should we check attributes for another name?
+
+            var fields = type.GetFields(); // public fields, only those can be serialized?
```

Comment:
I think we might want to use our existing reflection system to unify behavior with what's done for serialization. That way, it will skip/consider members the same way.
Here's how we can access them:
`TypeDescriptorFactory.Default.Find(typeof(XXX)).Members`